backend/src/main/java/com/redhat/cloud/notifications/routers/EndpointService.java

```diff
@@ -180,17 +200,58 @@ public Endpoint createEndpoint(@Context SecurityContext sec, @NotNull @Valid End
                 CamelProperties properties = endpoint.getProperties(CamelProperties.class);
                 String processorName = "p-" + endpoint.getAccountId() + "-" + UUID.randomUUID();
                 properties.getExtras().put(OB_PROCESSOR_NAME, processorName);
-                String processorId = setupOpenBridgeProcessor(endpoint, properties, processorName);
-
-                // TODO find a better place for these, that should not be
-                //       visible to users / OB actions
-                //       See also CamelTypeProcessor#callOpenBridge
-                properties.getExtras().put(OB_PROCESSOR_ID, processorId);
+                try {
+                    processorId = setupOpenBridgeProcessor(endpoint, properties, processorName);
+                    endpoint.setStatus(EndpointStatus.PROVISIONING);
+
+                    // TODO find a better place for these, that should not be
+                    //       visible to users / OB actions
+                    //       See also CamelTypeProcessor#callOpenBridge
+                    properties.getExtras().put(OB_PROCESSOR_ID, processorId);
+                    needChecking = true;
+                } catch (WebApplicationException we) {
+                    we.printStackTrace();
+                    endpoint.setStatus(EndpointStatus.FAILED);
+                }
 
             }
         }
 
-        return resources.createEndpoint(endpoint);
+        Endpoint createdEndpoint = resources.createEndpoint(endpoint);
+        if (obEnabled && needChecking && createdEndpoint.getSubType().equals("slack")) {
+            triggerReadyCheck(createdEndpoint.getId(), processorId);
+        }
+        return createdEndpoint;
+    }
+
+
+    private void triggerReadyCheck(UUID endpointId, String processorId) {
+
+        JobDataMap jdm = new JobDataMap();
+        jdm.put("epid", endpointId);
+        jdm.put("pid", processorId);
+        JobDetail jd = JobBuilder.newJob().ofType(ReadyCheck.class)
+                .withIdentity("myJob")
+                .usingJobData(jdm)
+                .build();
+
+        Trigger trigger = TriggerBuilder.newTrigger()
+                .withIdentity(new TriggerKey("myJobs", "obWorker"))
```

Comment:
I don't know how quartz work, is this the key for the trigger? won't other slacks  endpoints overwrite this job/trigger?

Reply:
I was also thinking of passing a unique name instead of `myJobs` for each scheduler.
For the moment I am glad this works at all :-)

Reply:
Wouldn't it be simpler to have one single `@Scheduled`-annotated method that would check the status of all endpoints that currently have a non-final status at period execution time?

Reply:
```
@Scheduled
void checkEndpointsStatus() {
  // Get from DB all endpoints in non-final status
  // Call OB for each endpoint
  // Update endpoint status in DB if needed
}
```

Reply:
Wouldn't it cause conflicts with the replicas?

Reply:
Not if Postgres' `PESSIMISTIC_LOCK` and `SKIP LOCKED` are both used.

Each endpoint in non-final status would only be processed by one pod at a time.

Reply:
Good question about a single job. I expect those provisioning/updating scenarios to be rather rare, so I don't know how much such a continuously running job makes sense. I am thinking that it makes a DB trip every 10s "for nothing". But then this is mostly a single select which is not too heavy. I don't know

Reply:
The impact on the DB should be reasonable enough if the status field is indexed.